src/class-pull-hours-harvester.php

```diff
@@ -121,97 +136,55 @@ private function backup() {
+			$filename = sanitize_title_with_dashes( $target ) . '.json';
+			$this->write( $filename, json_encode( $data['values'] ) );
 		}
 	}
 
 	/**
-	 * This method returns the URL to a Google spreadsheet - not a specific
-	 * worksheet, but the overall entity. Because we end up pulling this
-	 * information twice in different formats, we have an argument to
-	 * identify each.
+	 * This method establishes a client, with associated token, that can read
+	 * information from the Google Sheets API (v4).
 	 *
-	 * @param string $key An identifying key to a Google spreadsheet.
-	 * @param string $format Either 'json' or 'tabletop'.
+	 * @link https://developers.google.com/sheets/api/quickstart/php
 	 */
-	private function lookup_base_url( $key, $format ) {
-		$url = 'https://spreadsheets.google.com/feeds/worksheets/' .
-			   $key .
-			   '/public/basic';
-		switch ( $format ) {
-			case 'json':
-				$url .= '?alt=json';
-				break;
-			case 'tabletop':
-				$url .= '?alt=json-in-script&callback=Tabletop.singleton.loadSheets';
-				break;
+	private function get_service() {
+		error_log( 'Defining Google Sheet Service...' );
+		$client = new \Google_Client();
+		$client->setApplicationName('Library_Hours_Harvester');
+		$client->setScopes(\Google_Service_Sheets::SPREADSHEETS_READONLY);
+		$client->setAuthConfig( plugin_dir_path( __FILE__) . '../auth/credentials.json');
+
+		// Load previously authorized token from a file.
+		// It appears that this token need not be valid, it just needs to be present.
+		$token_path = plugin_dir_path( __FILE__ ) . '../auth/token.json';
+		if ( file_exists( $token_path ) ) {
+			$access_token = json_decode( file_get_contents( $token_path ), true );
+			$client->setAccessToken( $access_token );
 		}
-		return $url;
-	}
 
-	/**
-	 * This method returns the URL to a specific worksheet inside a Google
-	 * spreadsheet.
-	 *
-	 * @param string $key An identifying key to the specific worksheet inside
-	 * a Google spreadsheet.
-	 */
-	private function lookup_child_url( $key ) {
-		$url = 'https://spreadsheets.google.com/feeds/list/' .
-			   $this->spreadsheet_key . '/' . $key .
-			   '/public/values' .
-			   '?alt=json-in-script&callback=Tabletop.singleton.loadSheet';
-		return $url;
+		$service = new \Google_Service_Sheets( $client );
+
+		return $service;
 	}
 
 	/**
-	 * This method identifies the key to a specific worksheet, which is the
-	 * last few characters in a retrieved URL, after the final slash.
+	 * This method plucks the name of a sheet out of general metadata for a
+	 * spreadsheet.
 	 *
-	 * @param string $url A URL to a Google spreadsheet worksheet.
+	 * @return string $name The name of a single sheet in a spreadsheet.
 	 */
-	private function set_sheet_key( $url ) {
-		$length = strlen( $url );
-		$position = strrpos( $url, '/' ) + 1;
-		return substr( $url, $position - $length );
+	private function pluck_sheet_name( $haystack ) {
```

Comment:
This method is, in fact used, but apparently CodeClimate cannot detect it because the reference is in an array inside an array_map function on [line 96](https://github.com/MITLibraries/mitlib-pull-hours/pull/5/files/791190dd33871665153fb1bdcbd87acc2544f955#diff-0fd533daef857de2139a0ff208361609R96).